airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -0,0 +1,199 @@
+from abc import ABC, abstractmethod
+from typing import Any, Mapping, Tuple, Dict, Union, Iterable
+
+import requests
+from base_python import AbstractSource, HttpAuthenticator, HttpStream, IncrementalStream, Stream, SimpleAuthenticator
+
+
+class StripeStream(HttpStream):
+    url_base = "https://api.stripe.com/v1/"
+
+    def get_next_page_token(self, decoded_response: Dict[str, Any]) -> Union[Dict[str, Any], None]:
+        if decoded_response['has_more'] and bool(decoded_response['has_more']):
+            if decoded_response['data'] and len(decoded_response['data']) > 0:
+                last_object_id = decoded_response['data'][-1]['id']
+                return {'starting_after': last_object_id}
+        # return None  # skip for faster testing
+
+    def parse_response(self, decoded_response: Dict) -> Iterable[Dict]:
+        if decoded_response['data']:
+            for record in decoded_response['data']:
+                yield record
+
+    def get_request_params(self, **kwargs):
+        # TODO remove this limit. just here to speed up testing
+        return {'limit': 1}
+
+
+class IncrementalStripeStream(StripeStream, IncrementalStream, ABC):
+    continuously_save_state = False  # Stripe returns most recently created objects first :(
+
+    @property
+    @abstractmethod
+    def cursor_field(self) -> str:
+        pass
+
+    def get_request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().get_request_params(stream_state=stream_state)
+        params.update({
+            'created': stream_state.get(self.cursor_field)
+        })
+        return params
+
+    def get_updated_state(self, current_state, latest_record) -> Dict[str, Any]:
+        return {
+            self.cursor_field: max(latest_record.get(self.cursor_field), current_state.get(self.cursor_field, 0))
+        }
+
+
+class Customers(IncrementalStripeStream):
+    cursor_field = 'created'
+
+    def path(self, **kwargs) -> str:
+        return "customers"
+
+
+class BalanceTransactions(IncrementalStripeStream):
+    cursor_field = 'created'
+    name = "balance_transactions"
+
+    def path(self, **kwargs) -> str:
+        return "balance_transactions"
+
+
+class Charges(IncrementalStripeStream):
+    cursor_field = 'created'
```

Comment:
Can these types of setters use this type of behavior instead:

```
@property
def cursor_field(self) -> List[str]:
    return 'created`
```


The current method of setting of these properties for the parent class to use feels a bit unnatural. I'm definitely not a "Python developer" though...

@keu do you have a preference for this sort of thing? Do you consider one more accessible than the other?

Reply:
@jrhizor I don't think it is user friendly, class attributes used instead properties very often mainly because it is convenient to have `constant`-like values